sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1142 @@
+    """Find the target neighbors of each data sample.
+
+    Parameters
+    ----------
+    X : array, shape (n_samples, n_features)
+        The training samples.
+
+    y : array, shape (n_samples,)
+        The corresponding (encoded) training labels.
+
+    n_neighbors : int
+        The number of target neighbors to select for each sample in X.
+
+    classes : array, shape (n_classes,), optional (default=None)
+        The non-singleton classes, encoded as integers in [0, n_classes).
+        If None (default), they will be inferred from ``y``.
+
+    **nn_kwargs : keyword arguments
+        Parameters to be passed to a :class:`neighbors.NearestNeighbors`
+        instance except from ``n_neighbors``.
+
+    Returns
+    -------
+    target_neighbors: array, shape (n_samples, n_neighbors)
+        The indices of the target neighbors of each sample.
+    """
+
+    target_neighbors = np.zeros((X.shape[0], n_neighbors), dtype=np.intp)
+
+    nn = NearestNeighbors(n_neighbors=n_neighbors, **nn_kwargs)
+
+    if classes is None:
+        classes = np.unique(y)
+
+    for class_id in classes:
+        ind_class, = np.where(y == class_id)
+        nn.fit(X[ind_class])
+        neigh_ind = nn.kneighbors(return_distance=False)
+        target_neighbors[ind_class] = ind_class[neigh_ind]
+
+    return target_neighbors
+
+
+def _find_impostors_blockwise(X_a, X_b, radii_a, radii_b,
+                              return_distance=False, block_size=8):
+    """Find (sample, impostor) pairs in blocks to avoid large memory usage.
+
+    Parameters
+    ----------
+    X_a : array, shape (n_samples_a, n_components)
+        Transformed data samples from class A.
+
+    X_b : array, shape (n_samples_b, n_components)
+        Transformed data samples from class B.
+
+    radii_a : array, shape (n_samples_a,)
+        Squared distances of the samples in ``X_a`` to their margins.
+
+    radii_b : array, shape (n_samples_b,)
+        Squared distances of the samples in ``X_b`` to their margins.
+
+    block_size : int, optional (default=8)
+        The maximum number of mebibytes (MiB) of memory to use at a time for
+        calculating paired squared distances.
+
+    return_distance : bool, optional (default=False)
+        Whether to return the squared distances to the impostors.
+
+    Returns
+    -------
+    imp_indices : array, shape (n_impostors,)
+        Unraveled indices of (sample, impostor) pairs referring to a matrix
+        of shape (n_samples_a, n_samples_b).
+
+    imp_distances : array, shape (n_impostors,), optional
+        imp_distances[i] is the squared distance between samples imp_row[i] and
+        imp_col[i], where
+        imp_row, imp_col = np.unravel_index(imp_indices, dims=(n_samples_a,
+        n_samples_b))
+    """
+
+    n_samples_a = X_a.shape[0]
+    bytes_per_row = X_b.shape[0] * X_b.itemsize
+    block_n_rows = int(block_size*1024*1024 // bytes_per_row)
```

Comment:
PEP8 violation ! :-P

Reply:
I just updated my pep8, but it does not catch this. Do you know why?

Reply:
No idea. As far as I know there should be space between the * operations to follow PEP8 convention.